design/one-pager-template-stacks.md

```diff
@@ -0,0 +1,344 @@
+
+Crossplane Stacks manage multiple CRDs. Template Stacks should be no different.
+
+Template definitions will need to express the template to render for each CRD that is managed. The GVR can be used as a map key to denote the CRD template being described.
+
+Each template needed to render a GVR should be provided (demonstrated below as a map), additionally a name that represents that intent of that template should distinguish each template (also demonstrated as a map). This name can then be reused to reference a template (see [Horizontal Composition](#Horizontal-Composition)).
+
+```yaml
+kind: Stack
+metadata:
+  name: CachingWebServiceStack
+spec:
+  templates:
+    redis.example.org/v1:
+      deployment: |
+        kind: Deployment
+        …
+          containers:
+          - name: redis-controller
+            image: example/redis-controller:{{.spec.redisVersion}}
+    cachingwebservice.example.org/v1:
+      cache: |
+        apiVersion: example.org/v1
+        kind: Redis
+        …
+        spec:
+          redisVersion: "5"
+      web: |
+        kind: Deployment
+        …
+          containers:
+          - name: nginx-controller
+            image: example/nginx-controller:{{.spec.nginxVersion}}
+--
+kind: CachingWebService
+spec:
+  nginxVersion: "1.17.4"
+```
+
+In the example above, distinct templates are defined for the Redis CRD and the CachingWebService. CachingWebService includes two templates, one of which relies on the other.
+
+The `nginxVersion` is defined in the `spec` of the `CachingWebService` instance, while `redisVersion` has been declared in the `cache` template for `cachingwebservice.example.org/v1`. The template variable has been passed into a descendent Stack.
+
+An additional form of composition and reuse is available by referencing Template Stack kinds within new Template Stacks. For example, `MegaWebsite` could include `CachingWebServer` in the `dependsOn` field of that Stack.
+
+**TODO(displague) update "discussed later" to the appropriate section**
+
+#### Using Reconciled Values as Template Variables
+
+Kubernetes resources typically set their state within a `status` field, which is often available as a Status Sub-Resource. In fact, CRD resources specifically support scale and status sub-resources. Access to this sub-resource can be granted independently from the resource itself.
+
+The template capabilities available to a Template Stack should not be restricted for use within the `spec` or other resource creation features.
+
+Template Stack managed resources should be able to manage their status, as any controller managed resource would. Arguably, the most useful values to expose through this status would be derived from resources created with templates.
+
+```yaml
+kind: Stack
+metadata:
+  name: StatusExampleStack
+spec:
+  template:
+    example.example.org/v1:
+      controller: |
+        kind: Deployment
+        spec:
+        …
+  templateStatus:
+    example.example.org/v1:
+      controller: |
```

Comment:
`controller` is an arbitrary map key in this case.  It is confusing. I'll use a word with less baggage.

Reply:
I see you changed `controller` as `templateA`. Thanks for that, however my point was a bit different. 

When I check the value for this arbitrary key in `template.example.example.org/v1.templateA`, I see a k8s `Deployment` as value and feel like its status would be updated due to `templateStatus.example.example.org/v1.templateA`

What I mean is, putting statusField under a key seems not necessary and confusing since we are updating status of, `example.example.org/v1` and not `Deployment`.

For example, if we are two templates as below, would it make any difference to write `templateA` or `templateB` in `templateStatus` ? 
```yaml
kind: Stack
metadata:
  name: StatusExampleStack
spec:
  template:
    example.example.org/v1:
      templateA: |
        kind: Deployment
        spec:
        …
      templateB: |
        kind: Service
        spec:
        …
  templateStatus:
    example.example.org/v1:
      templateA: |
        statusField: {{.some.value}}
```

...or, do you mean, `{{.some.value}}` coming from status of object defined as `templateA` ?